mythril/ether/soliditycontract.py

```diff
@@ -102,6 +102,13 @@ def get_source_info(self, address):
 
         index = helper.get_instruction_index(self.disassembly.instruction_list, address)
 
+        if index is None:
+            # File index for instruction address not found. But is there only *one* source file?
+            file_idx_set = {m.solidity_file_idx for m in self.mappings}
+            if len(file_idx_set) == 1:
+                # Yes, so use the since file mentioned.
+                index = file_idx_set.pop()
```

Comment:
this always returns 0 right?

Reply:
It probalby returns 0 - but I think it not wise to make that kind of assumption when it isn't necessary.

Semantically, if the address isn't explicitly found, and there is only one file index, then use that file index. That you some know that file indices start from 0 and that there coudn't have been two file indices where one having index 0 was deleted are additional assumptions.

Reply:
I see a typo "since file" -> single file index"

Reply:
oops , the index there means the address index in the instructions list.

Reply:
oops, the index there means the index of the instruction in the assembly code list.(and yep that is counter intuitive).